syntax = "proto3";

option csharp_namespace = "GrpcGreeter";

package greet;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply);

  rpc StreamFromClientSayHello (stream HelloRequest1) returns (HelloReply);
}

// The request message containing the user's name.
message HelloRequest1 {
  string tv_show_name = 1;
  int32 tv_show_year = 2;
  string episode_name = 3;
  int32 episode_season = 4;
  int32 episode_number = 5;
  string character_name = 6;
  string actor_name = 7;
}

message HelloRequest {
  string name = 1;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1;
}
